awswrangler/mysql.py

```diff
@@ -340,16 +361,33 @@ def to_sql(
                 df.reset_index(level=df.index.names, inplace=True)
             column_placeholders: str = ", ".join(["%s"] * len(df.columns))
             insertion_columns = ""
+            upsert_columns = ""
+            upsert_str = ""
             if use_column_names:
                 insertion_columns = f"({', '.join(df.columns)})"
+            if mode.lower().strip() == "upsert_duplicate_key":
+                upsert_columns = ", ".join(df.columns.map(lambda column: f"`{column}`=VALUES(`{column}`)"))
+                upsert_str = f" ON DUPLICATE KEY UPDATE {upsert_columns}"
             placeholder_parameter_pair_generator = _db_utils.generate_placeholder_parameter_pairs(
                 df=df, column_placeholders=column_placeholders, chunksize=chunksize
             )
+            sql: str
             for placeholders, parameters in placeholder_parameter_pair_generator:
-                sql: str = f"INSERT INTO `{schema}`.`{table}` {insertion_columns} VALUES {placeholders}"
+                if mode.lower().strip() == "upsert_replace_into":
+                    sql = f"REPLACE INTO `{schema}`.`{table}` {insertion_columns} VALUES {placeholders}"
+                else:
+                    sql = f"INSERT INTO `{schema}`.`{table}` {insertion_columns} VALUES {placeholders}{upsert_str}"
                 _logger.debug("sql: %s", sql)
                 cursor.executemany(sql, (parameters,))
             con.commit()
+            if mode.lower().strip() == "upsert_distinct":
+                temp_table = f"{table}_{''.join(random.choice(string.ascii_lowercase) for i in range(10))}"
```

Comment:
`temp_table = f"{table}_{uuid.uuid4().hex}"` is simpler and only requires one library